### PR TITLE
Checkout: Add actions/reducer for receipts

### PR DESCRIPTION
**Pull request by drewblaisdell**
Addresses part of #2235.

In order to decrease the size of the PR required to refactor the thank you page, I picked these commits out of #3270.

This PR adds actions/reducers/an endpoint call for fetching a single receipt. This is necessary because checkout is being refactored to fetch a receipt from the server when rendering the thank you page, instead of temporarily storing transaction data in local storage.

**Testing**
This PR does not change the behavior of Calypso. The only modifications that will be included in the build are the new endpoint call in `lib/undocumented` and three new action types in `state/action-types`. I included some tests for the new reducer.

- [x] Code

### CODE REVIEW

File: client/lib/wpcom-undocumented/lib/undocumented.js [hunk 1]
```diff
@@ -1868,6 +1868,12 @@ Undocumented.prototype.startExport = function( siteId, advancedSettings, fn ) {
 	}, advancedSettings, fn );
 };
 
+Undocumented.prototype.getReceipt = function( receiptId, fn ) {
```
Comment (scruffian) [inline client/lib/wpcom-undocumented/lib/undocumented.js]: Maybe this should live in `/client/lib/wpcom-undocumented/lib/me.js`?

File: client/state/receipts/Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+REPORTER ?= spec
```
Comment (scruffian) [inline client/state/receipts/Makefile]: Just an idea. We keep adding tests in subdirectories, and our tests keep getting slower to run. Could we add this to /client and start to build something that can run all the tests together? Probably should be for another PR, but it was just a thought.

**Comment by scruffian** [comment]
One small comment, but LGTM :+1:

File: client/lib/wpcom-undocumented/lib/undocumented.js (hunk 1 above)
Comment (drewblaisdell) [inline client/lib/wpcom-undocumented/lib/undocumented.js]: Good catch - I moved it.

**Comment by drewblaisdell** [comment]
Thanks @scruffian.
